src/components/input/DateSelector.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import DatePicker, {getToday} from 'react-native-modern-datepicker';
+
+type dateSelectorProps = {
+  setDate: (date: Date) => void;
+  date: Date;
+};
+
+const DateSelector = ({setDate, date}: dateSelectorProps) => {
+  const today = getToday();
+  const selectedDate =
+    date.getFullYear() + '/' + (date.getMonth() + 1) + '/' + date.getDate();
+
+  const parseDateString = (dateString: String): Date => {
+    const [year, month, day] = dateString.split('/').map(Number);
+    return new Date(year, month - 1, day); // month is 0-based in Date constructor
+  };
+
+  const handleDateChange = (newDate: String) => {
+    const parsedDate = parseDateString(newDate);
+    setDate(parsedDate);
```

Comment:
maybe cause infinite re-render loop

```suggestion
    if (parsedDate.getTime() !== date.getTime()) {
      setDate(parsedDate);
    }
```

Reply:
e45ad34